doc/docker.md

```diff
@@ -1,16 +1,102 @@
 # Running with Docker
 
+The advantage of using docker-compose is that it is configured to run several services that the Control Panel relies on, such as the PostgreSQL database and Redis. It initializes these ready for local development purposes.
+
+## Build
+
 To build the docker image, use the following command:
+
 ```sh
 docker build -t controlpanel .
 ```
 
-Running the app requires a PostgreSQL database, and this is provided with
-docker-compose:
+## .env file
+
+You need to create a `.env` file with the settings to enable it to connect to external services.
+
+You probably want it to pick up your 'data' AWS profile:
+```sh
+AWS_PROFILE=data
+```
+
+For testing purposes, you can simply connect to some services used for the dev environment. To do this, for the keys listed below, copy the matching lines from chart-env-config/dev/cpanel.yml and just change the format of each line to `KEY=value`.
+```sh
+# To log in with OIDC
+OIDC_*
+# To interact with AWS account IAM config
+AWS_DATA_ACCOUNT_ID
+AWS_COMPUTE_ACCOUNT_ID
+```
+
+Unless you're testing the slack feature, just disable it with this line:
+```sh
+SLACK_API_TOKEN=disabled
+```
+
+For more details of environment variable settings, refer to: [Environment Variables Reference](environment.md).
+
+## AWS setup
+
+Control Panel needs AWS credentials to change S3 bucket IAM permissions, according to user requests. When this runs in the cluster it can get them using kube2iam. But when you run locally you'll need to provide AWS credentials that give access to the mojanalytics account. Typically you'll use configure to use your Landing Account User and use AssumeRole to the mojanalytics account.
+
+You can test what AWS Account is currently configured on your command-line, like this:
+
+```sh
+$ pip install boto3
+$ python -c "import boto3; print(boto3.client('sts').get_caller_identity()['Arn'])"
+arn:aws:sts::593291632749:assumed-role/restricted-admin-data/botocore-session-1590188888
+```
```

Comment:
I see the following error from a cold start:

```
$ python -c "import boto3; print(boto3.client('sts').get_caller_identity()['Arn'])"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/ntoll/.virtualenvs/moj/lib/python3.8/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/ntoll/.virtualenvs/moj/lib/python3.8/site-packages/botocore/client.py", line 626, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ExpiredToken) when calling the GetCallerIdentity operation: The security token included in the request is expired
```

Not sure what to suggest here. It feels to me like this command is too early in the process, perhaps it should only work once all the other AWS related configuration (explained below) is finished.

Reply:
Dunno on this. Perhaps you could try this again with the revised AWS instructions?